File: be/sales/invoicing_access.py
"""Invoicing catalog feature gates."""

from __future__ import annotations

from settings.module_features import is_module_feature_enabled


def invoice_creation_allowed() -> bool:
    return is_module_feature_enabled('invoicing', 'invoice_creation')


def invoice_tracking_allowed() -> bool:
    return is_module_feature_enabled('invoicing', 'invoice_tracking')


def payment_tracking_allowed() -> bool:
    return is_module_feature_enabled('invoicing', 'payment_tracking')

File: fe/src/utils/invoicingAccess.js
import { isModuleFeatureEnabled } from './moduleFeatures';
import { isModuleEnabled } from './moduleSettings';

function invoicingModuleActive() {
  return isModuleEnabled('invoicing');
}

export function invoiceCreationAllowed() {
  return (
    invoicingModuleActive() &&
    isModuleFeatureEnabled('invoicing', 'invoice_creation', false)
  );
}

export function invoiceTrackingAllowed() {
  return (
    invoicingModuleActive() &&
    isModuleFeatureEnabled('invoicing', 'invoice_tracking', false)
  );
}

export function paymentTrackingAllowed() {
  return (
    invoicingModuleActive() &&
    isModuleFeatureEnabled('invoicing', 'payment_tracking', false)
  );
}
